src/app/components/apparatus-entry/apparatus-entry.component.ts

```diff
@@ -1,26 +1,59 @@
-import { Component, Input } from '@angular/core';
-import { map } from 'rxjs/operators';
-import { ApparatusEntry } from '../../models/evt-models';
+import { ChangeDetectionStrategy, Component, Input, OnInit, Optional, SkipSelf } from '@angular/core';
+import { map, shareReplay } from 'rxjs/operators';
+import { ApparatusEntry, GenericElement } from '../../models/evt-models';
 import { register } from '../../services/component-register.service';
 import { EVTModelService } from '../../services/evt-model.service';
-import { EditionlevelSusceptible } from '../components-mixins';
-
+import { ApparatusEntryDetailComponent } from './apparatus-entry-detail/apparatus-entry-detail.component';
 @Component({
   selector: 'evt-apparatus-entry',
   templateUrl: './apparatus-entry.component.html',
   styleUrls: ['./apparatus-entry.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 @register(ApparatusEntry)
-export class ApparatusEntryComponent extends EditionlevelSusceptible {
+export class ApparatusEntryComponent implements OnInit {
   @Input() data: ApparatusEntry;
 
+  public opened = false;
+  public isInsideAppDetail: boolean;
+  public nestedApps: ApparatusEntry[] = [];
+
   variance$ = this.evtModelService.appVariance$.pipe(
     map((variances) => variances[this.data.id]),
+    shareReplay(1),
   );
 
   constructor(
     private evtModelService: EVTModelService,
+    @Optional() @SkipSelf() private parentDetailComponent?: ApparatusEntryDetailComponent,
```

Comment:
Are these decorators needed?

Reply:
@Optional() it's necessary because normally an `apparatusEntry` doesn't have a parent which is an `ApparatusEntryDetailComponent`. This only applies to nested apps.
@SkipSelf seems useless (I can't understand why I added it)